diff --git a/sources/meta-freescale/recipes-bsp/barebox/barebox.inc b/sources/meta-freescale/recipes-bsp/barebox/barebox.inc
index 20e5a4f..a8177ed 100644
--- a/sources/meta-freescale/recipes-bsp/barebox/barebox.inc
+++ b/sources/meta-freescale/recipes-bsp/barebox/barebox.inc
@@ -16,13 +16,15 @@ BAREBOX_BINARY ?= "barebox.bin"
 BAREBOX_IMAGE ?= "barebox-${MACHINE}-${PV}-${PR}.bin"
 BAREBOXENV_BIN ?= "bareboxenv-${MACHINE}-${PV}-${PR}.bin"
 BAREBOX_SYMLINK ?= "barebox-${MACHINE}.bin"
+BAREBOX_SYMLINK2 ?= "barebox.bin"
 BAREBOXENV_SYMLINK ?= "bareboxenv-${MACHINE}.bin"
+BAREBOXENV_SYMLINK2 ?= "bareboxenv.bin"
 
 INSANE_SKIP_${PN} = "True"
 
 do_pre_configure() {
 	   cp ${WORKDIR}/defconfig ${S}/.config
-	   oe_runmake oldconfig
+	   oe_runmake -C ${S} oldconfig
 }
 addtask pre_configure before do_configure after do_patch
 
@@ -41,8 +43,9 @@ do_deploy () {
 	install ${S}/${BAREBOX_BINARY} ${DEPLOY_DIR_IMAGE}/${BAREBOX_IMAGE}
 
 	cd ${DEPLOY_DIR_IMAGE}
-	rm -f ${BAREBOX_SYMLINK}
+	rm -f ${BAREBOX_SYMLINK} ${BAREBOX_SYMLINK2}
 	ln -sf ${BAREBOX_IMAGE} ${BAREBOX_SYMLINK}
+	ln -sf ${BAREBOX_IMAGE} ${BAREBOX_SYMLINK2}
 
 	install -d ${STAGING_BINDIR_NATIVE}
 	cd ${S}
@@ -57,7 +60,8 @@ do_env_deploy() {
 	install ${S}/defaultenv/barebox_default_env ${DEPLOY_DIR_IMAGE}/${BAREBOXENV_BIN}
 
 	cd ${DEPLOY_DIR_IMAGE}
-	rm -f ${BAREBOXENV_SYMLINK}
+	rm -f ${BAREBOXENV_SYMLINK} ${BAREBOXENV_SYMLINK2}
 	ln -sf ${BAREBOXENV_BIN} ${BAREBOXENV_SYMLINK}
+	ln -sf ${BAREBOXENV_BIN} ${BAREBOXENV_SYMLINK2}
 }
 addtask env_deploy before do_build after do_deploy
